src/prefect/tasks/snowflake/snowflake.py

```diff
@@ -0,0 +1,110 @@
+import snowflake.connector as sf
+
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+class SnowflakeQuery(Task):
+    """
+    Task for executing a query against a snowflake database.
+
+    Args:
+        - account (str): snowflake account name, see snowflake connector
+             package documentation for details
+        - user (str): user name used to authenticate
+        - password (str): password used to authenticate
+        - database (str, optional): name of the default database to use
+        - schema (int, optional): name of the default schema to use
+        - role (str, optional): name of the default role to use
+        - warehouse (str, optional): name of the default warehouse to use
+        - query (str, optional): query to execute against database
+        - data (tuple, optional): values to use in query, must be specified using placeholder is query string
+        - autocommit (bool, optional): set to True to autocommit, defaults to None, which
+            takes snowflake AUTOCOMMIT parameter
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        account: str,
+        user: str,
+        password: str,
+        database: str = None,
+        schema: str = None,
+        role: str = None,
+        warehouse: str = None,
+        query: str = None,
+        data: tuple = None,
+        autocommit: bool = None,
+        **kwargs
+    ):
+        self.account = account
+        self.user = user
+        self.password = password
```

Comment:
You may want to replace this explicit password with a Prefect secret, so that users don't have to hardcode the value into the task. [Here](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/tasks/dropbox/dropbox.py#L45) is how that's handled in a Dropbox task -- users supply the name of the secret (either in `__init__` or as a run-time argument) and the `Secret` by that name is accessed when the task runs.

Reply:
Comment & a question.

Comment: Makes total sense.  If it's alright with you, I'd take the pattern from [S3Download](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/tasks/aws/s3.py) which mimics how vault stores secrets (ie 1 secret maps to all necessary information for login: username, password, and account).

Question: Is the plan to require use of Prefect's secret manager?  My initial thought here was to use the pattern in the postgres task, where the user passes raw credentials.  That way users can leverage whatever secret management infrastructure they have in place (be it Vault, AWS Secret Manager, Prefect's secret manager, etc).

Reply:
That's an excellent question. The intent is definitely *NOT* to require Prefect's secret manager.

However this does raise an interesting question for discussion @cicdw -- we don't want to encourage users to pass secrets from task to task, because there's always a possibility of the secret becoming cached as a task result. To @akravetz's point, we definitely also don't want to require Prefect secrets. Perhaps there's a way for us to make Prefect secrets more pluggable, so that we could use a Prefect secret to deliver the value at runtime but the actual retrieval of that secret could be any preferred infrastructure?

In light of that let's not hold up this great PR for this!

Reply:
Yea - that's an interesting point!  My gut reaction is that we can encourage people to subclass `Secret` and override the `get` method - this seems like the simplest approach.  To do so, we might introduce a secrets _base_ class that we also inherit from for Prefect Secrets.  Let's open an issue for this to discuss further.

Also, as an aside I didn't catch that the postgres tasks used plain passwords and have [opened an issue](https://github.com/PrefectHQ/prefect/issues/1345) to update that.